app/uk/gov/hmrc/channelpreferences/controllers/PreferenceController.scala

```diff
@@ -76,4 +83,100 @@ class PreferenceController @Inject()(
           s", groupId: '${event.groupId}', timeStamp: '${event.timeStamp}', event: '${event.event}'"))
     }
   }
+
+  // ----------------------
+
+  private def doConfirmItsa(passedBack_entityId: String, passedBack_itsaId: String, authToken_saUtr: Option[String])(
+    implicit hc: HeaderCarrier): Future[Result] =
+    entityResolverConnector
+      .resolveBy(passedBack_entityId)
+      .flatMap { resolvedEntity =>
+        if (resolvedEntity.itsa.isDefined) {
```

Comment:
*Suggestions / Minor (feel free to discard)*: It might be worth using pattern matching which gives you the option to use extractor:
```scala
(resolvedEntity.itsa, resolvedEntity.saUtr, authToken_saUtr) match {
  case (Some(itsa),_,_) if itsa == passedBack_itsaId => ...
  case (Some(itsa),_,_) => ...
  case (None,Some(entitySaUtr),Some(authSaUtr)) if entitySaUtr == authSaUtr => ...
etc..
}
```

Reply:
most def not minor ;)

Reply:
Ok, fine.
There will be Scala pattern match cases instead of if/else branches.
Agreed